src/fsharp/NameResolution.fs

```diff
@@ -2031,22 +2031,22 @@ let rec ResolveLongIndentAsModuleOrNamespaceOrStaticClass sink (atMostOne: Resul
                     let especs  = mspecs @ tspecs
                     if not especs.IsEmpty then 
                         especs 
-                        |> List.map (fun mspec ->
-                            let subref = modref.NestedTyconRef mspec
+                        |> List.map (fun espec ->
+                            let subref = modref.NestedTyconRef espec
                             if IsEntityAccessible amap m ad subref then
                                 notifyNameResolution subref id.idRange
-                                look (depth+1) subref rest
+                                look (depth+1) (allowStaticClasses && subref.IsModuleOrNamespace) subref rest
```

Comment:
I think we should allow it. Just need to add a check that the outer type is also static.

Reply:
Yes, done, thanks